### ISSUE TITLE
Selayer is right?

### ISSUE THREAD

**Issue by seven320** [StickCui/PyTorch-SE-ResNet#4]
in your se-layer you use nn.Conv2d, but paper says that (batch, channel, H, W) to (batch, channel), so 
in my opinion that nn.Conv2d module to be nn.Linear module.
~~~
class Selayer(nn.Module):

    def __init__(self, inplanes):
        super(Selayer, self).__init__()
        self.global_avgpool = nn.AdaptiveAvgPool2d(1)
        self.conv1 = nn.Conv2d(inplanes, inplanes / 16, kernel_size=1, stride=1)
        self.conv2 = nn.Conv2d(inplanes / 16, inplanes, kernel_size=1, stride=1)
        self.relu = nn.ReLU(inplace=True)
        self.sigmoid = nn.Sigmoid()

    def forward(self, x):

        out = self.global_avgpool(x)

        out = self.conv1(out)
        out = self.relu(out)

        out = self.conv2(out)
        out = self.sigmoid(out)

        return x * out

~~~

**Comment by StickCui** [comment]
> in your se-layer you use nn.Conv2d, but paper says that (batch, channel, H, W) to (batch, channel), so
> in my opinion that nn.Conv2d module to be nn.Linear module.
> 
> ```
> class Selayer(nn.Module):
> 
>     def __init__(self, inplanes):
>         super(Selayer, self).__init__()
>         self.global_avgpool = nn.AdaptiveAvgPool2d(1)
>         self.conv1 = nn.Conv2d(inplanes, inplanes / 16, kernel_size=1, stride=1)
>         self.conv2 = nn.Conv2d(inplanes / 16, inplanes, kernel_size=1, stride=1)
>         self.relu = nn.ReLU(inplace=True)
>         self.sigmoid = nn.Sigmoid()
> 
>     def forward(self, x):
> 
>         out = self.global_avgpool(x)
> 
>         out = self.conv1(out)
>         out = self.relu(out)
> 
>         out = self.conv2(out)
>         out = self.sigmoid(out)
> 
>         return x * out
> ```

Yes， In this case， nn.Conv2d is equal to nn.Linear.

**Comment by seven320** [comment]
ahh, I misunderstanding. kernel_size is 1 so it is same as nn.Linear. thank you for your replying!!
